lib/fastlyctl/commands/logging/bigquery.rb

```diff
@@ -0,0 +1,125 @@
+
+        puts
+        puts "Name".ljust(max["name"]) + " | " + "Dataset".ljust(max["dataset"]) + " | " + "Table".ljust(max["table"]) + " | " + "ProjectId".ljust(max["project_id"])
+        puts "-" * (max["name"] + max["dataset"] + max["table"] + max["project_id"])
+        config.each { |c| 
+            puts "%s | %s | %s | %s" % [c["name"].ljust(max["name"]), c["dataset"].ljust(max["dataset"]), c["table"].ljust(max["table"]), c["project_id"].ljust(max["project_id"])]
+        }
+        puts
+    end
+
+
+    def self.create(options)
+        puts "Creating bigquery log endpoint"
+        required_opts = ["name", "format_file", "user", "secret_key_file", "project_id", "dataset", "table" ]
+        
+        ensure_opts(required_opts,options)
+
+        parsed_key = parse_secret_key(options[:secret_key_file])
+
+        parsed_format = File.read(options[:format_file])
+
+        params = {}
+
+        id  = options[:service]
+        version = FastlyCTL::Fetcher.get_writable_version(id) unless options[:version]
+        version ||= options[:version]
+
+        params[:name] = options[:name] 
+        params[:format] = parsed_format 
+        params[:user] = options[:user]
+        params[:secret_key] = parsed_key
+        params[:project_id] = options[:project_id]
+        params[:dataset] = options[:dataset]
+        params[:table] = options[:table]
+
+        FastlyCTL::Fetcher.api_request(:post, "/service/#{id}/version/#{version}/logging/bigquery", params: params)
+        puts "BigQuery logging provider created in service id #{id} on version #{version}"
+    end
+
+    def self.update(options)
+        required_opts = ["name"]
+        ensure_opts(required_opts,options)
+        
+        puts "Updating bigquery log endpoint #{options[:name]}"
+
+        parsed_key = parse_secret_key(options[:secret_key_file]) unless options[:secret_key_file].nil?
+        parsed_format = File.read(options[:format_file]) unless options[:format_file].nil?
+
+        params = {}
+
+        id  = options[:service]
+        version = FastlyCTL::Fetcher.get_writable_version(id) unless options[:version]
+        version ||= options[:version]
+
+        params[:name] = options[:new_name] unless options[:new_name].nil? 
+        params[:format] = parsed_format unless options[:format_file].nil?
+        params[:user] = options[:user] unless options[:user].nil?
+        params[:secret_key] = parsed_key  unless options[:secret_key_file].nil?
+        params[:project_id] = options[:project_id] unless options[:project_id].nil?
+        params[:dataset] = options[:dataset] unless options[:dataset].nil?
+        params[:table] = options[:table] unless options[:table].nil?
+
+        puts "Params #{params}"
+        FastlyCTL::Fetcher.api_request(:put, "/service/#{id}/version/#{version}/logging/bigquery/#{options[:name]}", params: params)
+        puts "BigQuery logging provider update in service id #{id} on version #{version}"
+    end
+
+    def self.list(options)
```

Comment:
We're not handling the case where the version is specified by argument.

Reply:
fixed.